src/main/webapp/js/PoemData.js

```diff
@@ -0,0 +1,123 @@
+import { CreateElement } from './Common.js';
+/**
+ * Gets a poem and its annotations
+ */
+class PoemData {
+  constructor() {
+    this.poemId = null;
+    this.poemData = null;
+    this.annotationData = null;
+  }
+  /**
+   * Get the poem data
+   * @returns Poem data as a JSON object
+   */
+  async GetPoemData() {
+    try {
+      const response = await fetch(`/poem?id=${this.poemId}`);
+      const poemDataJson = await response.json();
+      this.poemData = poemDataJson;
+      return poemDataJson;
+    } catch (err) {
+      throw new Error(err);
+    }
+  }
+  /**
+   * Get all the poem annotations
+   * @returns Annotation data as a JSON object
+   */
+  async GetPoemAnnotations() {
+    try {
+      const response = await fetch(`/annotations?poemId=${this.poemId}`);
+      const annotationsDataJson = await response.json();
+      this.annotationData = annotationsDataJson;
+      return annotationsDataJson;
+    } catch (err) {
+      throw new Error(err);
+    }
+  }
+  AddPoemToPage(poem) {
+    // Get the poem content container
```

Comment:
This is a really long function. Might be possible to clean it up with a few functions